src/main/java/com/hida/ArkMinter.java

```diff
@@ -0,0 +1,251 @@
+        this.AMOUNT = amount;
+        this.PREFIX = getIdPrefix();
+        this.FORMAT = getCharMap();
+        this.LENGTH = (10 - this.PREFIX.length());
+
+        /*
+        deciding whether or not to store these values in a database or just 
+        constantly generate them. If the latter, should I store them 
+        somewhere else, such as in a separate class or in the minter        
+        */
+        this.tokenMaps.put("DIGIT", "0123456789");
+        this.tokenMaps.put("LOWERCASE", "abcdefghijklmnopqrstuvwxyz");
+        this.tokenMaps.put("UPPERCASE", "ABCDEFGHIJKLMNOPQRSTUVWXYZ");
+        this.tokenMaps.put("MIXCASE", 
+                "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ");
+        this.tokenMaps.put("EXTENDED", "0123456789abcdefghijklmnopqrstuvwxyz");
+        this.tokenMaps.put("ALPHANUMERIC", 
+                "0123456789abcdefghijklmnopqrstuvwxyz"
+                        + "ABCDEFGHIJKLMNOPQRSTUVWXYZ");
+    }
+
+    /* constructors; currently unsed - subject to removal */
+    
+    public ArkMinter(int amount, int length) {
+        this.PREFIX = "";
+        this.AMOUNT = amount;
+        this.LENGTH = length;
+        this.FORMAT = "auto";
+    }
+
+    public ArkMinter(int amount, String prefix, int length) {
+        this.PREFIX = prefix;
+        this.AMOUNT = amount;
+        this.LENGTH = length;
+        this.FORMAT = "auto";
+    }
+
+    public ArkMinter(int amount, String format) {
+        this.PREFIX = "";
+        this.AMOUNT = amount;
+        this.LENGTH = format.length();
+        this.FORMAT = format;
+    }
+
+    public ArkMinter(int amount, String prefix, String format) {
+        this.PREFIX = prefix;
+        this.AMOUNT = amount;
+        this.LENGTH = (prefix.length() + format.length());
+        this.FORMAT = format;
+    }
+
+    /**
+     * * genIDauto() Method description: This generates an automated ID format.
+     * User would give a string that excepts 6 letters: {d, l, u, m, e, a}.
+     * Similar to the Mode settings in genIDauto() method. d:
+     *
+     * <p>
+     * DIGIT: {0123456789} Digit values only.</p>
+     * <p>
+     * LOWERCASE: {abc...xyz} Lowercase letters only. </p>
+     * <p>
+     * UPPERCASE: {ABC...XYZ} Uppercase letters only.</p>
+     * <p>
+     * MIXCASE: {abc...XYZ} Lowercase and Uppercase letters only. </p>
+     * <p>
+     * EXTENDED: {012...xyz} Digit values and Lowercase letters only. </p>
+     * <p>
+     * ALPHANUMERIC: {012...abc...XYZ} Digit values, Lowercase and Uppercase
+     * letters.</p>
+     *
+     * @param token: There are 2 different modes to choose from. RANDOM: Value
+     * by value, each value's order is randomized. SEQUENTIAL: Value by value,
+     * each value's order is sequenced.
+     * @return - A reference to a JsonObject that contains Json list of ids
+     */
+    @Override
+    public JsonObject genIdAuto(String token) {
+        String characters = (String) this.tokenMaps.get(token);

```

Comment:
This is very error prone since `token` is not guaranteed to be a valid key in `tokenMaps`.  The better way to handle this is to have an `enum` of the different token types instead:

```java
enum TokenType {
    MIXCASE,
    ALPHANUMERIC,
    ...
}
```